drivers/platform/aducm3029/rtc.c

```diff
@@ -0,0 +1,186 @@
+
+/******************************************************************************/
+/************************* Include Files **************************************/
+/******************************************************************************/
+
+#include <stdlib.h>
+#include "error.h"
+#include "rtc.h"
+#include "rtc_extra.h"
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+/**
+ * @brief Initialize the RTC peripheral.
+ * @param device - The RTC descriptor.
+ * @param init_param - The structure that contains the RTC initialization.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t rtc_init(struct rtc_dev **device, struct rtc_init_param *init_param)
+{
+	int32_t ret;
+	struct rtc_dev *dev;
+	struct aducm_rtc_dev *adev;
+	struct aducm_rtc_init_param *ainit = init_param->extra;
+
+	dev = (struct rtc_dev *)calloc(1, sizeof(*dev));
+	if (!dev)
+		return FAILURE;
+
+	adev = (struct aducm_rtc_dev *)calloc(1, sizeof(*adev));
+	if (!adev)
+		goto error_dev;
+
+	adev->memory = calloc(1, ADI_RTC_MEMORY_SIZE);
+	if(!adev->memory)
+		goto error_extra;
+
+	dev->id = init_param->id;
+	adev->prescaler = ainit->prescaler;
+	dev->extra = adev;
+
+	ret = adi_rtc_Open(dev->id, adev->memory, ADI_RTC_MEMORY_SIZE,
+			&adev->instance);
+	if(ret != SUCCESS)
+		goto error_mem;
+
+	ret = adi_rtc_EnableInterrupts(adev->instance, dev->id, true);
+	if(ret != SUCCESS)
+		goto error_open;
+
+	ret = adi_rtc_SetPreScale(adev->instance, adev->prescaler);
+	if(ret != SUCCESS)
+		goto error_open;
+
+	*device = dev;
+
+	return SUCCESS;
+
+error_open:
+	adi_rtc_Close(adev->instance);
+error_mem:
+	free(adev->memory);
+error_extra:
+	free(adev);
+error_dev:
+	free(dev);
+
+	return FAILURE;
+}
+
+/**
+ * @brief Free the resources allocated by rtc_init().
+ * @param dev - The RTC descriptor.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t rtc_remove(struct rtc_dev *dev)
+{
+	int32_t ret;
+	struct aducm_rtc_dev *adev = dev->extra;
+
+	ret = adi_rtc_Close(adev->instance);
+	if(ret != SUCCESS)
+		return FAILURE;
+
+	free(adev->memory);
+	free(adev);
+	free(dev);
+
+	return SUCCESS;
+}
+
+/**
+ * @brief Start the real time clock.
+ * @param dev - The RTC descriptor.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t rtc_start(struct rtc_dev *dev)
+{
+	struct aducm_rtc_dev *adev;
+
+	if (!dev)
+		return FAILURE;
+	adev = dev->extra;
+
+	return adi_rtc_Enable(adev->instance, true);
+}
+
+/**
+ * @brief Stop the real time clock.
+ * @param dev - The RTC descriptor.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t rtc_stop(struct rtc_dev *dev)
+{
+	struct aducm_rtc_dev *adev;
+
+	if (!dev)
+		return FAILURE;
+	adev = dev->extra;
+
+	return adi_rtc_Enable(adev->instance, false);
```

Comment:
are these error codes compatibe to our api ?

Reply:
At the moment, yes.